### ISSUE TITLE
Unused part of sequence expression is not flagged by `no-unused-expressions`

### ISSUE THREAD

**Issue by brettz9** [eslint/eslint#5533]
**What version of ESLint are you using?**

Whatever version is currently in the ESLint demo.

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**

The ESLint demo.

**What did you do? Please include the actual source code causing the issue.**

Add the example from `no-sequences` docs:

```js
var a = (3, 5);
```

**What did you expect to happen?**

To get `no-unused-expressions` errors.

**What actually happened? Please include the actual, raw output from ESLint.**

No expected error (only get a `no-unused-vars` error).

**Comment by platinumazure** [comment]
Confirmed in online demo.

Not sure what correct behavior is. I would expect `no-sequences` for sure, but `no-unused-expressions` maybe not since the value is being assigned.

**Comment by brettz9** [comment]
But the "3" value is not assigned, and it by itself is an expression.

**Comment by ilyavolodin** [comment]
I have a feeling that `no-sequences` might not be flagging anything inside parenthesis. I'm not sure why that would be the case, though. @michaelficarra any ideas?

**Comment by platinumazure** [comment]
@brettz9 I'm not sure I agree with that. By that logic, all sequences would also have to have a `no-useless-expressions` violation as well and that would just mean the rules overlap.

Incidentally, here are some other test cases from the online demo:

1. Code: `(3, 5);`
    * Result: `no-useless-expressions` shows up (expected IMO), `no-sequences` does not (buggy IMO)
1. Code: `3, 5;`
    * Result: `no-useless-expressions` shows up (expected IMO)`, no-sequences` shows up (expected IMO).

I think @ilyavolodin is right about parentheses being the difference here.

**Comment by brettz9** [comment]
@platinumazure : I would not expect all sequences to automatically get the `no-useless-expressions`, e.g., in the following:

```js
foo = (doSomething(), doSomethingElse());
```

...though in this case, I see it is currently not getting flagged by `no-sequences` in the demo.

In the following case, I think the `!!test`, while being an expression, should--unlike the "3" in the `(3, 5)` example--not be treated as "unused" because it ends up having consequences within the condition evaluation:

```js
if ((doSomething(), !!test));
```

**Comment by brettz9** [comment]
(Filed #5535 to add an option to prevent parentheses from disabling `no-sequences`.)

**Comment by michaelficarra** [comment]
If `no-unused-expressions` would flag left operand of `,`, we'd need an exception for the indirect eval pattern `(0, eval)("...")`.

**Comment by BYK** [comment]
Note sure if we still need to address this after #5566.

@michaelficarra, @brettz9, @nzakas, @ilyavolodin - thoughts?

**Comment by nzakas** [comment]
This is not a bug, it is working as designed. I don't think we should do anything here.
